runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -70,23 +77,60 @@ dependencies {
   shadow library.java.jackson_annotations
   shadow library.java.jackson_databind
   shadow library.java.slf4j_api
-  testCompile library.java.hamcrest_core
-  testCompile library.java.junit
-  testCompile project(":sdks:java:io:google-cloud-platform").sourceSets.test.output
-  testCompile project(path: ":sdks:java:core", configuration: "shadowTest")
-  testCompile project(":sdks:java:extensions:google-cloud-platform-core").sourceSets.test.output
-  testCompile library.java.guava_testlib
-  testCompile library.java.slf4j_jdk14
-  testCompile library.java.mockito_core
-  testCompile library.java.google_cloud_dataflow_java_proto_library_all
-  testCompile library.java.datastore_v1_protos
-  testCompile library.java.jackson_dataformat_yaml
+  shadowTest library.java.hamcrest_core
+  shadowTest library.java.junit
+  shadowTest project(":sdks:java:io:google-cloud-platform").sourceSets.test.output
+  shadowTest project(path: ":sdks:java:core", configuration: "shadowTest")
+  shadowTest project(":sdks:java:extensions:google-cloud-platform-core").sourceSets.test.output
+  shadowTest library.java.guava_testlib
+  shadowTest library.java.slf4j_jdk14
+  shadowTest library.java.mockito_core
+  shadowTest library.java.google_cloud_dataflow_java_proto_library_all
+  shadowTest library.java.datastore_v1_protos
+  shadowTest library.java.jackson_dataformat_yaml
+  validatesRunner project(path: ":sdks:java:core", configuration: "shadowTest")
+  validatesRunner project(path: project.path, configuration: "shadow")
 }
 
 test {
   systemProperties = [ "beamUseDummyRunner" : "true" ]
 }
 
+task validatesRunnerTest(type: Test) {
+  group = "Verification"
+  def dataflowProject = project.findProperty('dataflowProject') ?: 'apache-beam-testing'
+  def dataflowTempRoot = project.findProperty('dataflowTempRoot') ?: 'gs://temp-storage-for-validates-runner-tests/'
+  systemProperty "beamTestPipelineOptions", JsonOutput.toJson([
+          "--runner=TestDataflowRunner",
+          "--project=${dataflowProject}",
+          "--tempRoot=${dataflowTempRoot}",
+  ])
+
+
+  classpath = configurations.validatesRunner
+  testClassesDirs = files(project(":sdks:java:core").sourceSets.test.output.classesDirs)
+  useJUnit {
+    includeCategories 'org.apache.beam.sdk.testing.ValidatesRunner'
+    excludeCategories 'org.apache.beam.sdk.testing.LargeKeys$Above10MB'
```

Comment:
I don't see any excludes defined here: https://github.com/apache/beam/blob/a2a5d3d7aa59b5cfde1c47a6286bcb3ccd7f8c85/runners/google-cloud-dataflow-java/pom.xml#L64

Is there another place where you got this list from?

Reply:
I got it from here: https://github.com/apache/beam/blob/a2a5d3d7aa59b5cfde1c47a6286bcb3ccd7f8c85/runners/google-cloud-dataflow-java/pom.xml#L283

Happy to remove them, if I misread the pom.

On a separate note, I see I missed the parallelism -- it's executing sequentially (which was useful for testing it), but is not what we want in Jenkins.

Reply:
Sorry that I didn't see that and good catch on the 4 forks.